src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -0,0 +1,309 @@
+     */
+    private function fixCase(Tokens $tokens, $casePosition)
+    {
+        $empty = true;
+        $fallThrough = true;
+        $commentPosition = null;
+        for ($i = $tokens->getNextTokenOfKind($casePosition, [':', ';']) + 1, $max = count($tokens); $i < $max; ++$i) {
+            if ($tokens[$i]->isGivenKind([T_SWITCH, T_IF, T_ELSE, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_DO, T_FUNCTION, T_CLASS])) {
+                $empty = false;
+                $i = $this->getStructureEnd($tokens, $i);
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind([T_BREAK, T_CONTINUE, T_RETURN])) {
+                $fallThrough = false;
+
+                continue;
+            }
+
+            if ($tokens[$i]->equals('}') || $tokens[$i]->isGivenKind(T_ENDSWITCH)) {
+                if (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if ($this->isNoBreakComment($tokens[$i])) {
+                $commentPosition = $i;
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind([T_CASE, T_DEFAULT])) {
+                if (!$empty && $fallThrough) {
+                    if (null !== $commentPosition && $tokens->getPrevNonWhitespace($i) !== $commentPosition) {
+                        $this->removeComment($tokens, $commentPosition);
+                        $commentPosition = null;
+                    }
+
+                    if (null === $commentPosition) {
+                        $this->insertCommentAt($tokens, $i);
+                    } else {
+                        $this->ensureNewLineAt($tokens, $commentPosition);
+                    }
+                } elseif (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if (!$tokens[$i]->isGivenKind([T_COMMENT, T_WHITESPACE])) {
+                $empty = false;
+            }
+        }
+    }
+
+    /**
+     * @param Token $token
+     *
+     * @return bool
+     */
+    private function isNoBreakComment(Token $token)
+    {
+        if (!$token->isComment()) {
+            return false;
+        }
+
+        return preg_match('~^((//|#)\s*no break\s*)|(/\*\*?\s*no break\s*\*/)$~', $token->getContent());
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $casePosition
+     */
+    private function insertCommentAt(Tokens $tokens, $casePosition)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        $newlinePosition = $this->ensureNewLineAt($tokens, $casePosition);
+
+        $newlineToken = $tokens[$newlinePosition];
+
+        $nbNewlines = substr_count($newlineToken->getContent(), $lineEnding);
+        if ($tokens[$newlinePosition - 1]->isGivenKind(T_OPEN_TAG) && false !== strpos($tokens[$newlinePosition - 1]->getContent(), $lineEnding)) {
```

Comment:
Well, it works fine so I think it's ok to keep it.

IMO we should implement a new token transformer to extract the whitespace from `T_OPEN_TAG`, this would simplify logic here a lot. But it would be a BC break so not going to happen soon.